control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2210,23 +2210,26 @@ func (r *HostedControlPlaneReconciler) reconcileOIDCRouteResources(ctx context.C
 		if err != nil {
 			return fmt.Errorf("unable to parse issuer URL: %s", hcp.Spec.IssuerURL)
 		}
+
+		if route.CreationTimestamp.IsZero() {
+			route.Spec.Host = u.Host
```

Comment:
The field [is immutable](https://github.com/openshift/api/blob/master/route/v1/types.go#L19) so updating it (e.g. accidentally) would lead to repeated failure.